src/apps/geonetwork/src/main/java/org/geonetwork/config/GeoNetworkSsoConfiguration.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * (c) 2003 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license,
+ * available at the root application directory.
+ */
+package org.geonetwork.config;
+
+import java.util.HashMap;
+import java.util.Map;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.Setter;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+@ConfigurationProperties(prefix = "geonetwork.sso")
```

Comment:
Should we move this to geonetwork.security.oauth2 to make it relates to spring.security.oauth2 ? cf. https://github.com/geonetwork/geonetwork/blob/main/config/application.yml#L30

Reply:
ok, moved to this, and I changed the name of the config item so it was a bit more clear.

```
geonetwork:
  security:
    oauth2:
      registration:
        github:
          email: email
          name: login
          organization: company
          surname: lastname
        custom:
          email: email
          name: first_name
          organization: company
          surname: last_name
```

Also, fixed issue with the anonymous user.